tools/RAiDER/utilFcns.py

```diff
@@ -532,3 +532,84 @@ def writeWeatherVars2HDF5(lat, lon, x, y, z, q, p, t, proj, outName=None):
         T = f.create_dataset('t', data=t)
 
         f.create_dataset('Projection', data=proj.to_json())
+
+
+
+# Part of the following UTM and WGS84 converter is borrowed from https://gist.github.com/twpayne/4409500
+# Credits go to Tom Payne
+
+_projections = {}
+
+
+def zone(coordinates):
+    if 56 <= coordinates[1] < 64 and 3 <= coordinates[0] < 12:
+        return 32
+    if 72 <= coordinates[1] < 84 and 0 <= coordinates[0] < 42:
+        if coordinates[0] < 9:
+            return 31
+        elif coordinates[0] < 21:
+            return 33
+        elif coordinates[0] < 33:
+            return 35
+        return 37
+    return int((coordinates[0] + 180) / 6) + 1
+
+
+def letter(coordinates):
+    return 'CDEFGHJKLMNPQRSTUVWXX'[int((coordinates[1] + 80) / 8)]
+
+
+def project(coordinates):
+    z = zone(coordinates)
+    l = letter(coordinates)
+    if z not in _projections:
+        _projections[z] = pyproj.Proj(proj='utm', zone=z, ellps='WGS84')
+    x, y = _projections[z](coordinates[0], coordinates[1])
+    if y < 0:
+        y += 10000000
+    return z, l, x, y
+
+
+def unproject(z, l, x, y):
+    if z not in _projections:
+        _projections[z] = pyproj.Proj(proj='utm', zone=z, ellps='WGS84')
+    if l < 'N':
+        y -= 10000000
+    lng, lat = _projections[z](x, y, inverse=True)
+    return (lng, lat)
+
+def WGS84_to_UTM(lon, lat):
+    shp = lat.shape
+    lon = np.ravel(lon)
```

Comment:
```suggestion
    Z = lon.flatten() # .flatten() takes care of raveling and copying in one step. 
```

Reply:
You would do the same thing for X, Y

Reply:
@jlmaurer This is okay, because lon / lat can be changed (i.e. don't need to copy; returning the same pointer of the array is okay). While Z,L,X,Y are just initialized as new arrays with the same shape of lon (or lat). So X and Y are not typo, either.